mobile/src/main/java/com/alexstyl/specialdates/util/Notifier.java

```diff
@@ -89,19 +95,66 @@ public void forDailyReminder(ContactEvents events) {
                 );
 
         String title = NaturalLanguageUtils.joinContacts(context, events.getContacts(), 3);
-        String fullText = TextUtils.join(", ", events.getContacts());
 
         NotificationCompat.Builder builder = new NotificationCompat.Builder(context)
                 .setSmallIcon(R.drawable.ic_stat_contact_event)
                 .setContentTitle(title)
                 .setLargeIcon(largeIcon)
-                .setStyle(new NotificationCompat.BigTextStyle().bigText(fullText))
                 .setVisibility(NotificationCompat.VISIBILITY_PRIVATE)
                 .setAutoCancel(true)
-                .setContentText(fullText)
                 .setContentIntent(intent)
+                .setNumber(events.size())
                 .setColor(context.getResources().getColor(R.color.main_red));
 
+        if (events.size() == 1) {
+            ContactEvent event = events.getEvent(0);
+            String msg = "";
+
+            if (EventType.BIRTHDAY.equals(event.getType())) {
+                int age = event.getContact().getBirthday().getAgeOnYear(event.getYear());
+                msg = context.getString(R.string.turns_age, age);
+
+            } else if (EventType.NAMEDAY.equals(event.getType())) {
+                msg = context.getString(R.string.nameday);
+            }
+
+            NotificationCompat.BigTextStyle bigTextStyle = new NotificationCompat.BigTextStyle().bigText(msg);
+            bigTextStyle.setBigContentTitle(title);
+            builder.setContentText(msg);
+
+            builder.setStyle(bigTextStyle);
+
+        } else if (events.getContacts().size() > 1) {
+            NotificationCompat.InboxStyle inboxStyle = new NotificationCompat.InboxStyle();
+            inboxStyle.setBigContentTitle(title);
+
+            for (int i = 0; i < events.size(); ++i) {
+
+                ContactEvent event = events.getEvent(i);
+                Contact contact = event.getContact();
+                String name = contact.getDisplayName().toString();
+
+                String lineFormatted = name;
+
+                if (EventType.BIRTHDAY.equals(event.getType())) {
```

Comment:
❌ This is hard to extend. What happens when more events are added in the future? We would have to check all bits of code that do checks with specific events.
Also, a Birthday [might not include a year of birth](https://github.com/alexstyl/Memento-Namedays/blob/master/mobile/src/main/java/com/alexstyl/specialdates/contact/Birthday.java#L12), which means that the following bit will throw an exception if the contact does not have a specified year of birth.

You could just copy-paste the [`getLabelFor()`](https://github.com/alexstyl/Memento-Namedays/blob/master/mobile/src/main/java/com/alexstyl/specialdates/upcoming/ui/ContactEventView.java#L58) method. It checks only for that special case, and fallbacks to using the [label of the eventType](https://github.com/alexstyl/Memento-Namedays/blob/master/mobile/src/main/java/com/alexstyl/specialdates/events/EventType.java#L24) if there is no special case.

Don't worry about the code duplication from the other class. I'll solve it nicely in a later PR :)